experimental/examples/bubblebench/editable-shared-tree/src/schema.ts

```diff
@@ -0,0 +1,234 @@
+    name: brand("String"),
+    extraLocalFields: emptyField,
+    value: ValueSchema.String,
+});
+
+export const numberSchema = namedTreeSchema({
+    name: brand("number"),
+    extraLocalFields: emptyField,
+    value: ValueSchema.Number,
+});
+
+export const iBubbleSchema = namedTreeSchema({
+    name: brand("Test:BubbleBenchAppStateiBubble-1.0.0"),
+    localFields: {
+        x: fieldSchema(FieldKinds.value, [numberSchema.name]),
+        y: fieldSchema(FieldKinds.value, [numberSchema.name]),
+        r: fieldSchema(FieldKinds.value, [numberSchema.name]),
+        vx: fieldSchema(FieldKinds.value, [numberSchema.name]),
+        vy: fieldSchema(FieldKinds.value, [numberSchema.name]),
+    },
+    extraLocalFields: emptyField,
+});
+
+export const iBubbleSequenceSchema = namedTreeSchema({
+    name: brand("Test:BubbleBenchAppStateiBubbleSequence-1.0.0"),
+    localFields: {
+        [EmptyKey]: fieldSchema(FieldKinds.sequence, [iBubbleSchema.name]),
+    },
+    extraLocalFields: emptyField,
+});
+
+export const iClientSchema = namedTreeSchema({
+    name: brand("Test:BubbleBenchAppStateiClient-1.0.0"),
+    localFields: {
+        clientId: fieldSchema(FieldKinds.value, [stringSchema.name]),
+        color: fieldSchema(FieldKinds.value, [stringSchema.name]),
+        bubbles: fieldSchema(FieldKinds.sequence, [iBubbleSequenceSchema.name]),
+    },
+    extraLocalFields: emptyField,
+});
+
+export const iClientSequenceSchema = namedTreeSchema({
+    name: brand("Test:BubbleBenchAppStateiClientSequence-1.0.0"),
+    localFields: {
+        [EmptyKey]: fieldSchema(FieldKinds.sequence, [iClientSchema.name]),
+    },
+    extraLocalFields: emptyField,
+});
+
+export const AppStateSchema = namedTreeSchema({
+    name: brand("Test:BubbleBenchAppState-1.0.0"),
+    localFields: {
+        clients: fieldSchema(FieldKinds.sequence, [iClientSequenceSchema.name]),
+    },
+    extraLocalFields: emptyField,
+});
+
+export type BubbleTreeProxy = EditableTree & {
+    x: number;
+    y: number;
+    vx: number;
+    vy: number;
+    r: number;
+};
+
+export type ClientTreeProxy = EditableTree & {
+    clientId: string;
+    color: string;
+    bubbles: BubbleTreeProxy[];
+};
+
+export type AppStateTreeProxy = EditableTree & {
+    clients: ClientTreeProxy[];
+};
+
+export const rootAppStateSchema = fieldSchema(FieldKinds.value, [AppStateSchema.name]);
+
+export const AppStateSchemaData: SchemaData = {
+    treeSchema: new Map([
+        [stringSchema.name, stringSchema],
+        [numberSchema.name, numberSchema],
+        [iBubbleSchema.name, iBubbleSchema],
+        [iBubbleSequenceSchema.name, iBubbleSequenceSchema],
+        [iClientSchema.name, iClientSchema],
+        [iClientSequenceSchema.name, iClientSequenceSchema],
+        [AppStateSchema.name, AppStateSchema],
+    ]),
+    globalFieldSchema: new Map([
+        [rootFieldKey, rootAppStateSchema],
+    ]),
+};
+
+export const mockAppStateJsonTree: JsonableTree = {
```

Comment:
this does not seem like it belongs in this file since its not schema.

Reply:
Removed in latest commit.